.github/workflows/Check_BOM.yml

```diff
@@ -0,0 +1,32 @@
+name: Check BOM in CSV files
+
+on: workflow_call
+
+permissions:
+  contents: read
+
+jobs:
+  check-bom:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check for BOM in CSV files
+        run: |
+          shopt -s globstar
+          shopt -s nocaseglob
+          for file in **/*.csv; do
+            bom=$(hexdump -n 3 -e '/1 "%02x"' "$file")
+            if [ "$bom" = "efbbbf" ]; then
+              echo "Error: File $file has a BOM"
+              echo "::error file=$file::File has BOM"
+              bom_found=true
+            else
+              echo "File $file is valid"
+            fi
```

Comment:
Read the first 3 bytes of the file and compare them with the UTF-8 BOM representation `EF BB BF`
https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8